### DISCUSSION TITLE
BeforeValidate auto-generate field data for Array Field

### DISCUSSION THREAD

**Discussion by sksar** [payloadcms/payload#458] [Q&A]
Hello,
I am creating an Inventory Management System with Payload CMS.

I have a collection called purchase order that looks like this
```

export default {
    slug: 'purchaseOrders',
    fields: [
        {
            name: 'items',
            type: 'array',
            fields: [
                {
                    type: 'row',
                    fields: [
                        {
                            name: 'item', type: 'relationship', relationTo: 'items', hasMany: false,
                            admin: { width: '30%' },
                        },
                        {
                            name: 'buyPrice', type: 'number', required: true,
                            admin: { width: '10%' },
                        },
                        {
                            name: 'quantity', type: 'number', required: true,
                            admin: { width: '10%' },
                        },
                        {
                            name: 'amount', type: 'number',
                            admin: { width: '10%', readOnly: true },
                            hooks: {
                                beforeValidate: [(args) => {
                                    const { data } = args;
                                    console.log(data);
                                    // return ???
                                }]
                            }
                        }
                    ]
                } 
            ],
            minRows: 1,
        }
    ]
}
```

In the above what I want to achieve is a repeating structure of `Item - Buy Price - Quantity - Amount`
The amount should be calculated from the `BuyPrice * Quantity`.

But my problem is that in the `args.data`, I am getting all the data as a whole. And not the siblingFields.

So my question would be how do I scope the function or data to the row, so that I get the corresponding data. 
Or how can I get the index of the function call, because in the dev console I can see that the same data is getting logged twice. (which maybe is the expected behaviour). So I can use that index to now do the scope from the array maybe.

I am stuck at this for a long time, I have searched the docs for hours, I could not figure it out in any way.

**Comment by jmikrut** [comment]
Hey @sksar — very good question. 

We just published a beta (`payload@0.14.27-beta.0`) that includes a new `siblingData` argument to field hooks. 

[Here is the commit](https://github.com/payloadcms/payload/commits/master) which includes changes to docs, but the docs changes won't hit our website until the next upcoming minor release.

Give it a shot - this should be exactly what you need! Thank you for the great request!
